src/SyntheticNetworks.jl

```diff
@@ -179,10 +211,15 @@ function grow!(
                 to G."""
             if rand() <= q
                 i = rand(1:nv(graph))
-                dist_spatial =
-                    map(j -> euclidean(graph.vertexpos[i], graph.vertexpos[j]), 1:nv(graph))
-                l_edge = Step_G34(graph, i, dist_spatial, r)
+                dist_spatial = map(j -> euclidean(graph.vertexpos[i], graph.vertexpos[j]), 1:nv(graph))
+                l_edge = Step_G34(graph, i, dist_spatial, r, methods[types])
+                if l_edge == 0
+                    n_actual -= 1
```

Comment:
what is happening here? are you sure resetting the loop index does not cause problems?

Reply:
My idea was to redraw a node, if there isn't any node that the new node can be connected to. But I see now that there is definitely a problem, because I don't remove the added vertex, edge and type.  
```suggestion

                    pop!(types)
                    rem_vertex!(graph,nv(graph))
                    rem_edge!(graph, nv(graph),min_dist_vertex)
                    n_actual -= 1

```
But this should be okay, I think.

Reply:
I tried out the suggestion above, but I get graphs smaller than I wanted. I'm not sure what would make more sense to do if no node is found.

Reply:
You connect a new node always in step G2, so it only means that there is no additional link added.
Sometimes there might just be no possibility. The resulting graph should havve the same number of nodes but less edges.

We could, however, also redraw the node location and try again (limited to `k` trials, which is another parameter then). This can be done in the following way: You first draw random numbers and determine which steps have to be performed. then you execute the respective steps, returning the new edges if possible or redraw the location up to `k` times (you can use `try/catch` for that maybe.) Only then, edges are added.